src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -894,9 +951,21 @@ private void WriteOpcPackageProperties(ZipArchive package, string psmdcpPath)
         }
 
         // Generate a relationship id for compatibility
-        private string GenerateRelationshipId()
+        private string GenerateRelationshipId(string path)
         {
-            return "R" + Guid.NewGuid().ToString("N").Substring(0, 16);
+            if (_deterministic)
```

Comment:
This is a non-breaking change, so I'd just always calculate the id from hash.

Reply:
It would add processing time for packing when deterministic hasn't been opted into. I would prefer not to change anything about the non-deterministic builds.

Reply:
The extra overhead is likely insignificant. More deterministic build is always better. It would also reduce the amount of cases we need to test and maintain.

Note - all Roslyn compiler code is always deterministic, except for a few cases when the new deterministic behavior would be a breaking change.